### PR TITLE
Expose Lucenes Ukrainian analyzer

### LINKED ISSUE #19433: Expose Ukrainian analyzer
When upgrading to Lucene 6.2, we should expose the newly added Ukrainian analyzer

https://issues.apache.org/jira/browse/LUCENE-7287

### PR DESCRIPTION
**Pull request by cbuescher**
Since Lucene 6.2. the UkrainianMorfologikAnalyzer is available through the lucene-analyzers-morfologik jar. This change exposes the analyzer to be used as an elasticsearch plugin.

Closes #19433

### CODE REVIEW

File: plugins/analysis-ukrainian/src/test/java/org/elasticsearch/index/analysis/UkrainianAnalysisTests.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+    public void testDefaultsUkranianAnalysis() throws IOException {
+        final TestAnalysis analysis = createTestAnalysis(new Index("test", "_na_"), Settings.EMPTY,
+                new AnalysisUkrainianPlugin());
+
+        Analyzer analyzer = analysis.indexAnalyzers.get("ukrainian").analyzer();
+        MatcherAssert.assertThat(analyzer, instanceOf(UkrainianMorfologikAnalyzer.class));
+    }
+
+//    public void testDefaultsUkrainianAnalysis() throws IOException {
```
Comment (nik9000) [inline plugins/analysis-ukrainian/src/test/java/org/elasticsearch/index/analysis/UkrainianAnalysisTests.java]: leftover?

**Comment by cbuescher** [comment]
Since I'm new to how the `licenses` directory looks like: I just mapped the dependency licenses for the morfologik-* jars to the Lucene NOTICE and LICENSE file since they seem to include the sections for the morfologik project. Hopre this is okay. After reading the original issue in https://issues.apache.org/jira/browse/LUCENE-7287 I also wasn't sure if the dictionary included in lucene-analyzers-morfologik has it's own license that we need to copy over or if the Lucene license and notice files cover that.

File: plugins/analysis-ukrainian/src/main/java/org/elasticsearch/index/analysis/ukrainian/UkrainianAnalyzerProvider.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.index.analysis.ukrainian;
```
Comment (cbuescher) [inline plugins/analysis-ukrainian/src/main/java/org/elasticsearch/index/analysis/ukrainian/UkrainianAnalyzerProvider.java]: Thanks @johtani. I don't mind putting all code into the `o.e.index.analysis` package, I think I followed some other analysis plugin layout. Does anybody else have an opinion on this? Maybe there was a reason for grouping analysis plugin code in sub-packages? I somehow doubt it and will move all the code of this PR into `o.e.index.analysis` if nobody objects.

Comment (cbuescher) [inline plugins/analysis-ukrainian/src/main/java/org/elasticsearch/index/analysis/ukrainian/UkrainianAnalyzerProvider.java]: Kuromoji has the *AnalyzerProvider under `o.e.index.analysis` but the Plugin class under `o.e.plugin.analysis.kuromoji`. Is there a reason for this? Same for Stempel.

**Comment by nik9000** [comment]
I don't think it makes a big difference which package the stuff is in but for a plugin of this size I'd just put it all in one package. And I'd try not to clash with a package in core. Though I don't think it is big deal either way.

**Review by nik9000 — APPROVED** [review]

**Comment by johtani** [comment]
I agree with nik9000 . my comment is just out of curiosity.

**Comment by cbuescher** [comment]
@jpountz this can go on master and 5.x as well, right?

**Comment by jpountz** [comment]
Correct.

**Comment by cbuescher** [comment]
Merged to 5.x with f0eade6e682309f5e09d39861574654adbac8640 and 42d4b60026a72ca3ca3ce3a6ad7d3c8d6a809674

**Pull request merged**